CONTRIBUTING.md

```diff
@@ -0,0 +1,330 @@
+
+- [ ] Code builds
+- [ ] Code passes `make style`
+- [ ] Update the `CHANGELOG` file:
+    - [ ] Is this a new feature that the users need to be aware of? New or updated example or miniapp?
+    - [ ] Does it make sense to create a new section in the CHANGELOG to group this with other related features?
+- [ ] Update the `INSTALL` file :
+    - [ ] Has a new optional library been added?
+       - [ ] Make sure the external library is licensed under LGPL, not GPL!
+    - [ ] Does `make` or `cmake` have a new target?
+    - [ ] Did the requirements or the installation process change? (rare)
+- [ ] Update README:
+    - [ ] Does GLVis have new keys/functionality that needs to be documented?
+    - [ ] Is this a really major feature that needs mentioning in the short summary here? (rare)
+- [ ] New examples:
+    - [ ] All sample runs at the top of the example work
+    - [ ] Update examples/makefile
+      - [ ] Add the example code to the appropriate SEQ_EXAMPLES and PAR_EXAMPLES variables.
+      - [ ] Add any files generated by it to the clean target.
+    - [ ] Update examples/CMakeLists.txt
+      - [ ] Add the example code to the ALL_EXE_SRCS variable.
+      - [ ] Make sure THIS_TEST_OPTIONS is set correctly for the new example.
+- [ ] List the new example in doc/CodeDocumentation.dox
+- [ ] Companion pull request for documentation in mfem/web repo
+   - [ ] Update or add miniapp-specific documentation, see e.g. the src/meshing.md and src/electromagnetics.md.
+   - [ ] Add the description, labels and screenshots in src/examples.md and and src/img.
+   - [ ] In examples.md, list the example under the appropriate categories, add new categories if necessary.
+- [ ] New miniapps:
+   - [ ] All sample runs at the top of the miniapp work
+   - [ ] Update top-level makefile and makefile in corresponding miniapp directory
+   - [ ] Update CMake build system
+      - [ ] Update the CMakeLists.txt file in miniapps/ directory, if the new miniapp is in a new directory.
+      - [ ] Add/update the CMakeLists.txt file in the new miniapp directory.
+      - [ ] Consider adding a new test for the new miniapp.
+   - [ ] List the new miniapp in doc/CodeDocumentation.dox
+   - [ ] Companion pull request for documentation in mfem/web repo
+     - [ ] Update or add miniapp-specific documentation, see e.g. the src/meshing.md and src/electromagnetics.md.
+     - [ ] Add the description, labels and screenshots in src/examples.md and and src/img in the mfem/web repo.
+     - [ ] The miniapps go at the end of the page, and are usually listed only under a specific "Application (PDE)" category.
+- [ ] New capability:
+   - [ ] All significant new classes, methods and functions have Doxygen-style documentation in source comments
+   - [ ] Consider adding new sample runs in existing examples to highlight the new capability
+   - [ ] Consider saving cool simulation pictures with the new capability in the Confluence gallery
```

Comment:
Submitting new images to the gallery was mentioned earlier and should be encouraged again here.  My only concern is that off-site users/developers may not have access to Confluence so we may want to make this bullet point more general.